file_transfer/src/file_transfer.c

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2016 HARBRICK TECHNOLOGIES, INC
+ *
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * \example file_transfer.c
+ *
+ * File Transfer Example.
+ *
+ * Demonstrates how to use the PolySync file transfer API to initiate and
+ * support file transfer operations between nodes.
+ *
+ */
+
+
+
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <signal.h>
+#include <semaphore.h>
+
+// PolySync API headers
+#include "polysync_core.h"
+#include "polysync_sdf.h"
+#include "polysync_message.h"
+#include "polysync_node.h"
+
+// Header for PolySync File Transfer API,
+// which is the topic of this example.
+#include "polysync_file_transfer.h"
+
+
+// All polysync-manager nodes support file transfer operations.
+// This example will attempt a file transfer with the first polysync-manager
+// it can find. This semaphore is used to signal when the first polysync-manager
+// has responded, and its GUID has been captured.
+sem_t manager_response;
+
+
+// After a polysync-manager node has been requested to provide its status,
+// it will respond. This is the name of the type of that message.
+const char manager_status_msg_name[] = "ps_response_msg";
+
+
+// When polysync-manager responds with its status,
+// this function will capture its GUID and signal 
+// that the program may proceeed via the semaphore.
+static void ps_manager_status_msg_handler(
```

Comment:
Not all the examples use the doxygen style I believe, and to me the doxygen style is more inhibitive than helpful in an example.